src/utils/roles.ts

```diff
@@ -0,0 +1,47 @@
+import { ColorResolvable, GuildMember, Role, RoleManager } from 'discord.js';
+
+export const addOrRemove = {
+  add: true,
+  remove: false,
+};
+
+const createRole = async (roleName: string, roles: RoleManager): Promise<Role> => {
+  try {
+    // create role object
+    const role = {
+      name: roleName,
+      color: 'GREY' as ColorResolvable,
+      reason: `AUTOMATED: Creating the role ${roleName}.`,
+    };
+    return await roles.create(role);
+  } catch (err) {
+    throw new Error(`Failed to create the role ${roleName}: ${err}`);
+  }
+};
+
+export const updateMemberRole = async (
+  member: GuildMember,
+  roleName: string,
+  add: boolean,
+): Promise<void> => {
+  let role = member.guild.roles.cache.find((role) => role.name === roleName);
```

Comment:
Again, use `roles.find. ...` rather than `roles.cache.find ...`

Reply:
Yup, I also fetched the roles here.